fsw/pc-linux/src/cfe_psp_memory.c

```diff
@@ -677,16 +677,31 @@ int32 CFE_PSP_InitProcessorReservedMemory( uint32 RestartType )
    if ( RestartType == CFE_PSP_RST_TYPE_PROCESSOR )
```

Comment:
CCB may want these to be OS_DEBUG so it has zero impact on current behavior unless debug is enabled (assuming this passes cppcheck).

Reply:
Adding #ifdef OS_DEBUG_PRINTF ... will still cause warning.

Reply:
Great point, didn't realize OS_DEBUG was a no-op if OS_DEBUG_PRINTF is not defined.  How about:
```
if (result != whatever_the_success_code_is)
  OS_printf("xyz_function didn't return success (%d)\n", return_code);
```

Reply:
I have several complaints about this code - we should clean it up properly rather than just squelching the cppcheck warnings about it.  After all, the point of running cppcheck is to actually improve the code, not just to keep the tool quiet.

Two issues:

1. the calls to the sub-functions are the same, in the same order, just with a different (hard-coded!) parameter, which should be equal to the `RestartType` value to begin with.  So can we just call e.g. 

`CFE_PSP_InitCDS(RestartType)`

and so on, and get rid of the `if` statement and `else` branch completely?

2. Failure of any of these memory initialization routines will almost certainly cause CFE to fail later on.  If any failure occurs, it should not just print it, but stop the process and return the value to the caller.  Also, the caller should check the value and do a `CFE_PSP_Panic()` if it fails.  Currently it ignores it.